node/test/Test/Pos/Client/Txp/Mode.hs

```diff
@@ -0,0 +1,291 @@
+         DB.MonadBlockDBGenericWrite (BlockHeader ssc) (Block ssc) Undo TxpTestInitMode where
+    dbPutBlund = DB.dbPutBlundPureDefault
+
+instance
+    (HasCoreConstants, SscHelpersClass ssc) =>
+    DB.MonadBlockDBGeneric (Some IsHeader) (SscBlock ssc) () TxpTestInitMode
+  where
+    dbGetBlock  = DB.dbGetBlockSscPureDefault @ssc
+    dbGetUndo   = DB.dbGetUndoSscPureDefault @ssc
+    dbGetHeader = DB.dbGetHeaderSscPureDefault @ssc
+
+----------------------------------------------------------------------------
+-- Boilerplate TxpTestContext instances
+----------------------------------------------------------------------------
+
+instance GS.HasGStateContext TxpTestContext where
+    gStateContext = ttcGState_L
+
+instance HasSlottingVar TxpTestContext where
+    slottingTimestamp = ttcSystemStart_L
+    slottingVar = GS.gStateContext . GS.gscSlottingVar
+
+instance HasLens GenesisWStakeholders TxpTestContext GenesisWStakeholders where
+    lensOf = ttcParams_L . ttpGenesisContext . gtcWStakeholders
+
+instance HasLens DB.DBPureVar TxpTestContext DB.DBPureVar where
+    lensOf = GS.gStateContext . GS.gscDB . pureDBLens
+      where
+        getter = \case
+            DB.RealDB _   -> realDBInTestsError
+            DB.PureDB pdb -> pdb
+        setter _ pdb = DB.PureDB pdb
+        pureDBLens = lens getter setter
+        realDBInTestsError = error "You are using real db in tests"
+
+----------------------------------------------------------------------------
+-- Boilerplate TxpTestMode instances
+----------------------------------------------------------------------------
+
+instance MonadGState TxpTestMode where
+    gsAdoptedBVData = DB.gsAdoptedBVDataDefault
+
+instance (HasCoreConstants, MonadSlotsData ctx TxpTestMode)
+      => MonadSlots ctx TxpTestMode
+  where
+    getCurrentSlot = do
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure (Just slot)
+    getCurrentSlotBlocking =
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotBlockingSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure slot
+    getCurrentSlotInaccurate = do
+        view ttcSlotId_L >>= \case
+            Nothing -> Slot.getCurrentSlotInaccurateSimple =<< view ttcSSlottingVar_L
+            Just slot -> pure slot
+    -- FIXME: this has already caused a bug with Explorer (CSE-203).
+    -- The workaround was to replace it with a `Timestamp 0`.
+    currentTimeSlotting = Slot.currentTimeSlottingSimple
+
+instance DB.MonadDBRead TxpTestMode where
+    dbGet = DB.dbGetPureDefault
+    dbIterSource = DB.dbIterSourcePureDefault
+
+instance MonadAddresses TxpTestMode where
+    type AddrData TxpTestMode = ()
+    getNewAddress _ = pure address
+      where
+        -- seed for address generation is a ByteString with 32 255's
+        seed = BS.replicate seedSize (255 :: Word8)
+        (_, address) = generateAddressAux seed
```

Comment:
Wait, so we do not own the related secret key for the change address? Don't we want to use the change later? How does it work?

Reply:
The change is never used in the tests.

Reply:
> Wait, so we do not own the related secret key for the change address?

Actually we do own, it's generated deterministically from constant seed.
But such kind of ownage is quite bad 😝 
Anyway, it doesn't matter if the change isn't used.